If-risk/InsuranceCompany.cs

```diff
@@ -0,0 +1,102 @@
+using If_risk.Exceptions;
+
+namespace If_risk;
+
+public class InsuranceCompany:IInsuranceCompany
+{
+    public string Name { get; }
+    public IList<Risk> AvailableRisks { get; set; }
+    private readonly IList<Policy> _policies;
+
+    public InsuranceCompany(string name, IList<Risk> availableRisks)
+    {
+        Name = name;
+        AvailableRisks = availableRisks;
+        _policies = new List<Policy>();
+    }
+    
+    public InsuranceCompany(string name, IList<Risk> availableRisks, IList<Policy> policyList)
+    {
+        Name = name;
+        AvailableRisks = availableRisks;
+        _policies = policyList;
+    }
+
+    public IPolicy SellPolicy(string nameOfInsuredObject, DateTime validFrom, short validMonths, IList<Risk> selectedRisks)
+    {
+        var validRiskNames = AvailableRisks.Select(x => x.Name).ToList();
+        var invalidRisks = selectedRisks.Where(x => !validRiskNames.Contains(x.Name)).ToList();
+        if (invalidRisks.Any())
+        {
+            throw new CompanyDoesntInsureRiskException(invalidRisks);
+        }
+
+        try
+        {
+            GetPolicy(nameOfInsuredObject, validFrom);
+            throw new PolicyAlreadyExistsException(nameOfInsuredObject, validFrom);
+        }
+        catch (PolicyNotFoundException)
+        {
+
+            _policies.Add(new Policy(nameOfInsuredObject, validFrom,
+                validFrom.AddMonths(validMonths) - TimeSpan.FromDays(1), selectedRisks));
+        }
+        
+        return _policies.Last();
```

Comment:
var gadīties ka atgriež vairākas polises?

Reply:
Last() atgriež tikai 1 pēdējo elementu, vai kā tu to domāji?

Reply:
viss cool.... nebija uzreiz uztverams kods sītais.. lokāli paskatījos... nav baigi labi man liekas šitā. bet fine...